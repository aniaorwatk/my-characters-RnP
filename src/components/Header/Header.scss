@import "../../scss/Main";

.header {
  position: relative;
  width: 100vw;
  height: 25vh;;
  padding: 10px;

  &__title {
    font-family: "Moon Dance", cursive;
    font-size: $font-size-XL;
    color: $color-text-white;
    text-shadow: 2px 2px 2px $color-text-black;
  }

  &__img{
    position: absolute;
    background-image: url("../../assets/diceHeader.jpg");
    background-position: center;
    background-size: cover;
  }
  .dark{
    top:0;
    left:70%;
    width: 15%;
    height: 100%;
    z-index: -1;
  }

  .light{
    top:0;
    left:0;
    width: 100vw;
    height: 100%;
    opacity: 0.3;
    z-index: -2;
  }
}



@media only screen and (min-width: $breakpointL) {
  .header {

    &__title {
      font-size: $font-size-h1BigHeader;
    }
  }
}
