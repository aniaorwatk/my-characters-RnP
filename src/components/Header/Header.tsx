import { labels } from "../../labels";
import "./Header.scss";

const Header = () => {
  return (
    <div className="header">
      <h1 className="header__title">{labels.header.title.toUpperCase()}</h1>
      <div className="header__img dark"></div>
      <div className="header__img light"></div>
    </div>
  );
};

export default Header;
