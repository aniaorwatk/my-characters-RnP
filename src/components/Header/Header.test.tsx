import { render, screen } from '@testing-library/react';
import Header from './Header';

test('renders Header with title', () => {
    render(< Header />);

    const title = screen.getByText("MY CHARACTERS IN PEN AND PAPER GAMES");

    expect(title).toBeInTheDocument();
});


test('renders snapshot', () => {
    render(< Header />);

    const title = screen.getByText("MY CHARACTERS IN PEN AND PAPER GAMES");

    expect(title).toMatchSnapshot();
});