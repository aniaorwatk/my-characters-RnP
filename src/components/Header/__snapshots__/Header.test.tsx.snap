// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<h1
  class="header__title"
>
  MY CHARACTERS IN PEN AND PAPER GAMES
</h1>
`;
